README.md

```diff
@@ -53,73 +85,87 @@ Once you save the file, bPanel will automatically install the plugins and rebuil
+Since node and wallet services are run on different servers,
+you will likely need different configurations to connect to the wallet. These
+should be in the same client conf file, prefaced with `wallet-` (note that bcoin looks for these
+in separate config files). See the sample conf file for an example.
 
-If you want to stop any of the containers, you can run the command:
+## About the Docker Environment
+There are three docker services in the compose file: `app`, `bcoin` and `securityc`.
+The `app` service is bPanel and acts as a static file server and as a request router
+to backend services. The `bcoin` service is an instance of `bcoin` that supports an http
+server, a wallet server and a bitcoin p2p server.
 
-```bash
-$ docker-compose stop [SERVICE_NAME]
-```
+The `securityc` service generates TLS keys and certs and runs a TLS terminating reverse proxy.
+You can use custom configs to connect to an existing node,
+or use the bcoin docker service to spin up a bcoin node that the webapp will connect to.
 
-This is useful if you do not want to develop with all of the services running.
+#### Configuration between Docker services
+These instructions are for if you want to run bPanel within the `app` service and have it talk to
+a bcoin node running in a container from the `bcoin` service. For example, this is how bPanel works
+out of the box if you simply run `docker-compose up -d`.
+
+Configurations are shared between the two docker containers using ENV files
+and the `environment` settings in the `docker-compose.yml`. If you are mounting
+a local bcoin data dir (`~/.bcoin`) or persisting using docker volumes, you can also pass settings to your bcoin docker container with a `bcoin.conf` file (read more about bcoin configurations [here](https://github.com/bcoin-org/bcoin/blob/master/docs/Configuration.md)).
+
+API keys can be shared through the `secrets.env` file. If you run `npm install` and
+there is no `secrets.env` present, one will automatically be generated for you with
+a cryptographically secure api key. The docker container environments will automaticaly set
+these variables via the `env_file` setting in `docker-compose`.
 
 ### Bcoin Setup Scripts
-Setup scripts are also supported. This will allow you to run scripts on your
+This section is only relevant if you will be running a bcoin node in a docker container
+using the `bcoin` service.
+
+This setup supports setup scripts. This will allow you to run scripts on your
 node for a repeatable and predictable environment for testing or development.
 
 Three circumstances need to be met to run a script:
 1. There needs to be a js file to run in the `scripts` directory that exports a function to run
 2. You need to pass the name of this file (including the extension)
-as an environment variable named `BCOIN_INIT_SCRIPT` in the docker-compose
+as an environment variable named `BCOIN_INIT_SCRIPT` in the docker-compose or as a `init-script`
+setting in your `bcoin.conf` file
```

Comment:
We should add something about the interface to the `init-script` - it should export a function that accepts a `node` and a `config`

Reply:
The following lines have some detail about passing the node. I updated to include the config and updated the wording a bit.